File: convert-to-pdf.py
"""
Convert README.html to PDF with clickable links
Requires: pip install weasyprint
"""

from weasyprint import HTML
import os

# Get the directory of this script
script_dir = os.path.dirname(os.path.abspath(__file__))

# Input and output paths
html_file = os.path.join(script_dir, 'README.html')
pdf_file = os.path.join(script_dir, 'README.pdf')

# Convert HTML to PDF
print("🔄 Converting README.html to PDF...")
HTML(filename=html_file).write_pdf(pdf_file)
print(f"✅ PDF created successfully: {pdf_file}")
print("📄 The PDF includes clickable links!")

File: convert-to-pdf.js
const puppeteer = require('puppeteer');
const path = require('path');

(async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();

    // Load the HTML file
    const htmlPath = path.join(__dirname, 'README.html');
    await page.goto(`file://${htmlPath}`, { waitUntil: 'networkidle0' });

    // Generate PDF with clickable links
    await page.pdf({
        path: 'README.pdf',
        format: 'A4',
        printBackground: true,
        margin: {
            top: '20px',
            right: '20px',
            bottom: '20px',
            left: '20px'
        }
    });

    console.log('✅ PDF created successfully: README.pdf');
    await browser.close();
})();
